src/makeCalculator.js

```diff
@@ -2,9 +2,34 @@
 
 /**
  * @return {object}
+ * - Methods: `add`, `subtract`, `multiply`, `divide`, `reset`, `operate`.
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    add(y, x) {
+      return y + x;
+    },
+    subtract(y, x) {
+      return y - x;
+    },
+    multiply(y, x) {
+      return y * x;
+    },
+    divide(y, x) {
+      return y / x;
+    },
+    operate(callback, param) {
+      this.result = callback(this.result, param);
```

Comment:
You can rewrite like this
```suggestion
      callback.call(this, param);
```